### PR TITLE
Enhance external image hosting in agent configuration

### LINKED ISSUE #4556: Unauthorized Use of External Image Hosting
## Description:

When creating an assistant, a user can set an avatar image from a library or upload one to Google's servers.

An attacker can intercept the assistant creation request and modify the "pictureUrl" field to include an image hosted on a different server or a link to a large file. This can result in abuse of the user's and server's bandwidth.

## Impact:

The use of external image hosting can lead to unauthorized bandwidth usage and potential denial-of-service (DoS) attacks on the Dust workspace and the remote server.

## Recommendations:

To mitigate this vulnerability, we should  implement the following measures:
- Verify that the "pictureUrl" field contains a URL from Google's servers.
- Verify that the content of "pictureUrl" is indeed an image.

### PR DESCRIPTION
**Pull request by flvndvd**
## Description

This PR fixes https://github.com/dust-tt/dust/issues/4556.

This PR addresses a security issue by improving the way we handle image uploads. It ensures that uploaded files have an image* mimeType, are less than 3MB in size, and restricts the accepted file types in the file picker.

In the agent configuration creation, it verifies that the file is self-hosted on our GCS and has the proper mime type, or it will yield an error.

To make this solution more resilient in the future, this PR starts centralizing the DFS (Distributed File System) logic in one single place.

It also applies some refactoring to the existing logic of content fragment, to use the newly introduced `DFS` class.

## Risk

Worst case it breaks avatar upload on agent configuration or the content fragment (uploading file, downloading file, ...), but it has been well tested locally.

### CODE REVIEW

File: front/lib/dfs/index.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
+import type { Bucket } from "@google-cloud/storage";
```
Comment (spolu) [inline front/lib/dfs/index.ts:1]: `gcs` rather than `dfs` in name no?

**Review by spolu — APPROVED** [review]
LGTM

I thiiiiiink content fragment could be moved as well to your new interface. Maybe worth doing it right now to convince ourselves it's the right one (esp as this PR is not to big)

**Comment by flvndvd** [comment]
> LGTM
> 
> I thiiiiiink content fragment could be moved as well to your new interface. Maybe worth doing it right now to convince ourselves it's the right one (esp as this PR is not to big)

I was planning to do it in a follow up PR, but let me bake this into this PR.

**Comment by spolu** [comment]
Up to you :+1:

File: front/lib/dfs/index.ts (hunk 1 above)
Comment (flvndvd) [reply front/lib/dfs/index.ts:1]: I stick with DFS for this file/class but all other occurrences will go with GCS.

**Review by spolu — APPROVED** [review]

**Pull request merged**